rule/expr_test.go

```diff
@@ -227,3 +236,100 @@ func TestValue(t *testing.T) {
 	require.False(t, v1.Equal(rule.BoolValue(false)))
 	require.False(t, v1.Equal(rule.StringValue("true")))
 }
+
+// A ComparableExpression can check its equivalence to another ComparableExpression
+func TestExprSame(t *testing.T) {
+	expr1 := rule.Eq(
+		rule.Int64Value(1),
+		rule.Int64Value(1),
+	).(rule.ComparableExpression)
+	expr2 := rule.Eq(
+		rule.Int64Value(1),
+		rule.Int64Value(1),
+	).(rule.ComparableExpression)
+	expr3 := rule.Eq(
+		rule.Int64Value(1),
+		rule.Int64Value(2),
+	).(rule.ComparableExpression)
+	expr4 := rule.And(
+		rule.BoolValue(true),
+		rule.BoolValue(true),
+	).(rule.ComparableExpression)
+
+	assert.True(t, expr1.Same(expr2))
+	assert.False(t, expr1.Same(expr3))
+	assert.False(t, expr1.Same(expr4))
+}
+
+// A ComparableExpression can check it equivalence to a complete AST.
+func TestExprTreeSame(t *testing.T) {
```

Comment:
🌷 Should we add a test case with a different operator?
For example:
```
expr5 := rule.Eq(
		rule.And(
			rule.And(
				rule.BoolValue(true),
				rule.BoolValue(true),
			),
			rule.Eq(
				rule.BoolValue(false),
				rule.BoolValue(true),
			),
			rule.BoolValue(true),
		),
		rule.BoolValue(false),
	).(rule.ComparableExpression)

assert.False(t, expr1.Same(expr5))

```

Reply:
Good call @YassineE , Done!